workers/unity/Packages/com.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -149,39 +68,53 @@ internal static IEnumerable<BuildTarget> GetUnityBuildTargets(SpatialBuildPlatfo
                 {
                     throw new Exception(IncompatibleWindowsPlatformsErrorMessage);
                 }
-
+                
                 result.Add(BuildTarget.StandaloneWindows);
-            }
-
-            if ((actualPlatforms & SpatialBuildPlatforms.Windows64) != 0)
+            } 
+            else if ((actualPlatforms & SpatialBuildPlatforms.Windows64) != 0)
             {
                 result.Add(BuildTarget.StandaloneWindows64);
             }
 
             return result.ToArray();
         }
+        
+        private static SpatialBuildPlatforms GetCurrentBuildPlatform()
+        {
+            switch (Application.platform)
+            {
+                case RuntimePlatform.WindowsEditor:
+                    return SpatialBuildPlatforms.Windows64;
+                case RuntimePlatform.OSXEditor:
+                    return SpatialBuildPlatforms.OSX;
+                case RuntimePlatform.LinuxEditor:
+                    return SpatialBuildPlatforms.Linux;
+                default:
+                    throw new Exception($"Unsupported platform detected: {Application.platform}");
+            }
+        }
 
-        private static void BuildWorkerForTarget(WorkerPlatform workerPlatform, BuildTarget buildTarget,
+        private static void BuildWorkerForTarget(string workerType, BuildTarget buildTarget,
             BuildOptions buildOptions, BuildEnvironment targetEnvironment)
         {
-            var spatialOSBuildConfiguration = GetBuildConfiguration();
-
             Debug.LogFormat("Building \"{0}\" for worker platform: \"{1}\", environment: \"{2}\"", buildTarget,
-                workerPlatform, targetEnvironment);
-
-            var symbols = PlayerSettings.GetScriptingDefineSymbolsForGroup(BuildTargetGroup.Standalone);
-
+                workerType, targetEnvironment);
+            
+            var spatialOSBuildConfiguration = SpatialOSBuildConfiguration.instance;
+            var buildTargetGroup = BuildPipeline.GetBuildTargetGroup(buildTarget);
+            var symbols = PlayerSettings.GetScriptingDefineSymbolsForGroup(buildTargetGroup);
+            
             try
-            {
-                var workerBuildData = new WorkerBuildData(workerPlatform, buildTarget);
-                var scenes = spatialOSBuildConfiguration.GetScenePathsForWorker(workerPlatform);
+            {   
+                var workerBuildData = new WorkerBuildData(workerType, buildTarget);
+                var scenes = spatialOSBuildConfiguration.GetScenePathsForWorker(workerType);
 
-                var typeSymbol = $"IMPROBABLE_WORKERTYPE_{workerBuildData.WorkerPlatformName.ToUpper()}";
+                var typeSymbol = $"IMPROBABLE_WORKERTYPE_{workerBuildData.WorkerType}";
```

Comment:
Can we strip this out? Afaik we don't use conditional compilation for different worker types

Reply:
🔥 burn it 🔥